### PR TITLE
Add cli features

### PR DESCRIPTION
**Pull request by MaHaWo**
- add remaining features to the cli. each wraps a function of the watcher that can be called via a cli command
   - change analyzer 
   - pause 
   - continue 
   - restart 
   - status 
   - cleanup 
- add `setup_info` function to make it easier to find out about a current setup 
- add implementations for basic `cmd.Cmd` methods: `emptyline`, `postloop`, `precmd`, `postcmd` 
- make sure the default config supplied by the models are used: Each model has its own default configuration file instead of one global one which then would have to be useful for all. The global one is only there for documentation purposes.
- refactor `do_start` function because there's some overlap with `change_analyzer`. This lead to the addition of helper functions

**bugfixes:** 
  - try to make sure setup tests use patched functions.  
  - fix issue in watcher test where the test didn't wait correctly for the watcher to stop 
  - add a queue to capture and report exceptions from the watcher process - these are not caught in the parent. could have been its own pr, but it's important here and not a particularly big change. This addresses the issue of the watcher processes throwing without the main process being able to know it.

roughly half the change lines of code are test functions for the repl.

### CODE REVIEW

**Comment by MaHaWo** [comment]
this includes stuff that is in #43 into the documentation. I'm not a good writer, so it should probably considered to be a first draft.

locally it can be build be executing `make html`  in the iSparrow/doc folder. it also has a [readthedocs project](https://readthedocs.org/projects/isparrow/) which is set to build on master pushes, however.

File: src/iSparrow/repl.py [hunk 1]
```diff
@@ -173,32 +250,67 @@ def wait_for_watcher_event(
...
+            "set_up: set up iSparrow for usage. Usage: 'set_up --cfg=<path>'. When no argumnet is provided, the default is used."
+        )
+        print(
+            "start: start a watcher for analyzing incoming files in a directory. Usage: 'start --cfg=<path>'. When no argumnet is provided, the default from birdnetlib is used."
+        )
+        print("stop: stop a previously started watcher")
+        print("pause: pause a running watcher")
+        print("continue: continue a paused watcher")
+        print("restart: restart an existing watcher")
+        print("change analyzer: change the analyzer of a running watcher")
```
Comment (iulusoy) [inline src/iSparrow/repl.py]: mention the argument that needs to be passed, and how

**Review by iulusoy — APPROVED** [review]

**Comment by iulusoy** [comment]
Also, the repo currently has 20 open branches, are they all necessary? I would aim to have at most 5 branches per person, to manage complexity.

**Comment by MaHaWo** [comment]
No, the branches are not necessary, and I deleted the ~15 or so oldest ones. We could delete them automatically upon merge in the future, but personally I like to keep them around for a bit longer...

File: src/iSparrow/repl.py (hunk 1 above)
Comment (MaHaWo) [reply src/iSparrow/repl.py]: Added respective hints in the overall help method, and added a utility function to retrieve the function docstring for each command upon request

**Pull request merged**